### PR TITLE
conf: check storage or coprocessor readpool config when partially unified

### LINKED ISSUE #7384: Configuration is not fully checked if the unified read pool is partially enabled
### What version of TiKV are you using?

Nightly or latest release-4.0.

### What operating system and CPU are you using?

Unrelated.

### Steps to reproduce

The following invalid configuration is not rejected:

```toml
[readpool.storage]
use-unified-pool = false
stack-size = "1MiB"

[readpool.coprocessor]
use-unified-pool = true
```

### What did you expect?

The above configuration is rejected because the minimal stack size is 2MiB.

### What did happened?

TiKV starts.

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

Issue Number: close #7384 

Problem Summary: When read pools are partially unified, invalid configurations may not be rejected.

### What is changed and how it works?

For storage or coprocessor read pools, check their own configurations when they're not unified.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

Fix the bug that an invalid storage or coprocessor read pool configuration may be not rejected when the unified thread pool is enabled.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1759,28 +1766,6 @@ mod readpool_tests {
 
     #[test]
     fn test_unified_enabled() {
-        // Allow invalid storage and coprocessor config when yatp is used.
```
Comment (sticnarf) [inline src/config.rs:1762]: It's deleted because we are now specifying whether to use the unified read pool in storage and coprocessor separately

**Comment by AndreMouche** [comment]
@BusyJay PTAL, thank you

**Pull request merged**